v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/SpannerChangeStreamsToBigQueryIT.java

```diff
@@ -0,0 +1,224 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.spannerchangestreamstobigquery;
+
+import static org.apache.beam.it.truthmatchers.PipelineAsserts.assertThatPipeline;
+import static org.apache.beam.it.truthmatchers.PipelineAsserts.assertThatResult;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.TableResult;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.teleport.metadata.SkipDirectRunnerTest;
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.apache.beam.it.common.PipelineLauncher.LaunchConfig;
+import org.apache.beam.it.common.PipelineLauncher.LaunchInfo;
+import org.apache.beam.it.common.PipelineOperator.Config;
+import org.apache.beam.it.common.PipelineOperator.Result;
+import org.apache.beam.it.common.utils.ExceptionUtils;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.gcp.TemplateTestBase;
+import org.apache.beam.it.gcp.bigquery.BigQueryResourceManager;
+import org.apache.beam.it.gcp.spanner.SpannerResourceManager;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Integration test for {@link SpannerChangeStreamsToBigQuery Spanner Change Streams to BigQuery}
+ * template.
+ */
+@Category({TemplateIntegrationTest.class, SkipDirectRunnerTest.class})
+@TemplateIntegrationTest(SpannerChangeStreamsToBigQuery.class)
+@RunWith(JUnit4.class)
+public class SpannerChangeStreamsToBigQueryIT extends TemplateTestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SpannerChangeStreamsToBigQueryIT.class);
+
+  private SpannerResourceManager spannerResourceManager;
+  private BigQueryResourceManager bigQueryResourceManager;
+
+  private static final AtomicInteger counter = new AtomicInteger(0);
```

Comment:
Not sure why counter and relevant methods are necessary when only row is being written

Reply:
Because we will have more complex test cases coming up which will write more than one rows in one test case(see comment above). Draft follow up pr: https://github.com/ChangyuLi28/DataflowTemplates/commit/a1c2872a700a410fed7c95193a0c4103a3fa55df#diff-28b3790277fb958cc3acef6775b2e789ad279e97d8f962068915c9800a647869

Let me know if you still prefer to remove this atomic counter and just use hard coded random uuid key value instead. Thanks!

Reply:
No that makes sense if you plan to test a chain of events in a single test case. I would still look at my comment above because you may be able to make that chaining easier, but in general a counter makes more sense than UUID because it guarantees to not have row collisions

Reply:
I see, to avoid further modifications I think it makes sense keep the increment counters in this base pr.